api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,34 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  /**
+   * Try to get the 'X-Access-Token' from the request headers.
+   * If no token is provided,
+   *   throw a 401 error with the message 'authentication required'
+   * Use jwt.verify() to verify the authenticity of the token and extract its payload
+   * Assign the extracted payload to the user property of the req object.
+   * Call next() (with no arguments) to let Express know to advance to its next route or middleware.
+   */
+
+  /**
+    * References:
+    * https://expressjs.com/en/4x/api.html#req.get
+    * https://nodejs.org/api/http.html#http_message_headers
+    * https://github.com/auth0/node-jsonwebtoken#jwtverifytoken-secretorpublickey-options-callback
+    */
+  const clientToken = req.get('X-Access-Token');
+  if (!clientToken) {
+    throw new ClientError(400, 'authentication required');
+  }
+  try {
+    const payload = jwt.verify(clientToken, process.env.TOKEN_SECRET);
+    req.user = payload;
+    next();
+  } catch (err) {
+    console.error(err);
+  }
```

Comment:
You don't have to use a `try catch` block here, but since you did you need to call `next` with the error object inside the `catch` block.
```JAVASCRIPT
catch (err) {
  next(err);
}
```

Reply:
thats right, we just did that exercise and went over this.... lol changes made